src/backend/vulkanll/src/lib.rs

```diff
@@ -45,14 +66,117 @@ pub struct SwapChain {
+    // Externally Synchronized Parameters: The `instance` parameter in `vkDestroyInstance`
+    // `Arc` ensures that we only call drop once
+    inner: Arc<InstanceInner>,
 }
 
 impl core::Instance for Instance {
     type B = Backend;
+
+    fn create() -> Instance {
+        // TODO: return errors instead of panic
+        let entry = ash::Entry::new().expect("Unable to load vulkan entry points");
+
+        let app_info = vk::ApplicationInfo {
+            s_type: vk::StructureType::ApplicationInfo,
+            p_next: ptr::null(),
+            p_application_name: "vulkan_ll".as_ptr() as *const _, // TODO:
+            application_version: 0,
+            p_engine_name: "gfx-rs".as_ptr() as *const _,
+            engine_version: 0, //TODO
+            api_version: 0, //TODO
+        };
+
+        let instance_extensions = entry.enumerate_instance_extension_properties()
+                                       .expect("Unable to enumerate instance extensions");
+        // Check our surface extensions against the available extensions
+        let surface_extensions = SURFACE_EXTENSIONS.iter().filter_map(|ext| {
+            instance_extensions.iter().find(|inst_ext| {
+                unsafe { CStr::from_ptr(inst_ext.extension_name.as_ptr()) == CStr::from_ptr(ext.as_ptr() as *const i8) }
+            }).and_then(|_| Some(*ext))
+        }).collect::<Vec<&str>>();
+
+        let instance = {
+            let cstrings = surface_extensions.iter()
+                                    .map(|&s| CString::new(s).unwrap())
+                                    .collect::<Vec<_>>();
+
+            let str_pointers = cstrings.iter()
+                                    .map(|s| s.as_ptr())
+                                    .collect::<Vec<_>>();
+
+            let create_info = vk::InstanceCreateInfo {
+                s_type: vk::StructureType::InstanceCreateInfo,
+                p_next: ptr::null(),
+                flags: vk::InstanceCreateFlags::empty(),
+                p_application_info: &app_info,
+                enabled_layer_count: 0,
+                pp_enabled_layer_names: ptr::null(),
+                enabled_extension_count: str_pointers.len() as u32,
+                pp_enabled_extension_names: str_pointers.as_ptr(),
+            };
+
+            entry.create_instance(&create_info, None).expect("Unable to create vulkan instance")
+        };
+
+        Instance {
+            inner: Arc::new(InstanceInner(instance)),
+        }
+    }
+
+    fn enumerate_physical_devices(&self) -> Vec<PhysicalDevice> {
+        self.inner.0.enumerate_physical_devices()
+            .expect("Unable to enumerate physical devices")
+            .iter()
+            .map(|&device| {
+                // TODO: add an ash function for this
+                let properties = unsafe {
+                    let mut out = mem::zeroed();
+                    self.inner.0.fp_v1_0().get_physical_device_properties(device, &mut out);
```

Comment:
Seems that I didn't wrap that function yet, I'll do it tomorrow. If you encounter that again tell me and I'll add them for you.